### PR TITLE
Improve? test reliability

### PR DESCRIPTION
**Pull request by ctz**
This is a variety of improvements aimed at debugging/fixing https://github.com/rustls/rustls-openssl-compat/actions/runs/8944014364/attempts/1

### CODE REVIEW

File: rustls-libssl/tests/runner.rs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 use std::io::Read;
 use std::process::{Child, Command, Output, Stdio};
-use std::{fs, net, thread, time};
```
Comment (cpu) [inline rustls-libssl/tests/runner.rs:3]: nit: I think `use std::sync::atomic` should be on its own.

File: rustls-libssl/tests/runner.rs [hunk 2]
```diff
@@ -31,6 +31,9 @@ use std::{fs, net, thread, time};
 #[test]
 #[ignore]
 fn client_unauthenticated() {
+    let port = choose_port();
+    let port_str = &format!("{port}");
```
Comment (cpu) [inline rustls-libssl/tests/runner.rs:36]: nit: I think this feels a little more natural as:

```rust
let port_str = &port.to_string();
```

Alternatively, maybe `choose_port()` should return `(u16, String)`? It seems like in all cases we end up also needing to `format!()` or `to_string()` a String as well.

**Review by cpu — APPROVED** [review]
:+1:

File: rustls-libssl/tests/runner.rs (hunk 2 above)
Comment (ctz) [reply rustls-libssl/tests/runner.rs:36]: > Alternatively, maybe `choose_port()` should return `(u16, String)`?

yeah that seems reasonable for test code.

**Pull request merged**